plugin/keyring_vault/vault_credentials_parser.cc

```diff
@@ -72,15 +76,41 @@ namespace keyring
   {
     reset_vault_credentials(&vault_credentials_in_progress);
 
-    std::ifstream credentials_file(file_url.c_str());
-    if (!credentials_file)
+    File_io file_io(logger);
+    File file = file_io.open(PSI_NOT_INSTRUMENTED,
+                             file_url.c_str(),
+                             O_RDONLY, MYF(MY_WME));
+
+    if (file < 0 || file_io.seek(file, 0, MY_SEEK_END, MYF(MY_WME)) == MY_FILEPOS_ERROR)
```

Comment:
The file must be closed if seek returns an error

Reply:
Done